storage/database/src/backend/heed/env.rs

```diff
@@ -282,10 +280,16 @@ where
         tx_rw: &RefCell<heed::RwTxn<'env>>,
     ) -> Result<impl DatabaseRw<T>, RuntimeError> {
         // Open up a read/write database using our table's const metadata.
-        Ok(HeedTableRw {
-            db: self.create_database(&mut tx_rw.borrow_mut(), Some(T::NAME))?,
-            tx_rw,
-        })
+        let db = {
+            let mut tx_rw = tx_rw.borrow_mut();
+            self.database_options()
+                .types()
+                .key_comparator()
+                .name(T::NAME)
+                .create(&mut tx_rw)?
+        };
+
+        Ok(HeedTableRw { db, tx_rw })
```

Comment:
## Key 2/4
... `heed` requires this info at runtime (kinda). We can actually do this at compile time.
Opening/creating a table now:
- Creates a database builder
- Sets the correct key/value types
- Sets the correct sort order

The user does not have to call a "set sort order" function, they define their sort order at compile time with `trait Key`, which forwards to...